notifyme-sdk-backend/notifyme-sdk-backend-ws/src/main/java/ch/ubique/notifyme/sdk/backend/ws/insertmanager/insertfilters/BeforeOnsetFilter.java

```diff
@@ -0,0 +1,39 @@
+package ch.ubique.notifyme.sdk.backend.ws.insertmanager.insertfilters;
+
+import ch.ubique.notifyme.sdk.backend.model.UserUploadPayloadOuterClass;
+import ch.ubique.notifyme.sdk.backend.ws.insertmanager.InsertException;
+import ch.ubique.notifyme.sdk.backend.ws.insertmanager.OSType;
+import ch.ubique.notifyme.sdk.backend.ws.security.NotifyMeJwtRequestValidator;
+import ch.ubique.notifyme.sdk.backend.ws.semver.Version;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * When a person tests positive, the "onset" date since when they are infectious is estimated.
+ * Visits which occurred before the onset are dropped.
+ */
+public class BeforeOnsetFilter implements UploadInsertionFilter {
```

Comment:
Either in this Filter or in a specific filter we have to check that no checkins for the future are uploade - this would otherwise lead to dangerous attack possibilites!

Reply:
Resolved: See [https://github.com/SwissCovid/swisscovid-cn-backend/commit/f2c7f5d7d0485263e0a8cb4f20a33d30f36edf4f](https://github.com/SwissCovid/swisscovid-cn-backend/commit/f2c7f5d7d0485263e0a8cb4f20a33d30f36edf4f)
Note that the entire repository has been moved to [SwissCovid/swisscovid-cn-backend](https://github.com/SwissCovid/swisscovid-cn-backend) and future work will continue there.